### PR TITLE
Hw 41

### PR DESCRIPTION
**Pull request by TimoxaV**

### CODE REVIEW

File: src/main/java/com/cinema/dao/impl/OrderDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    @Override
+    public List<Order> getUsersOrders(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("from Order o join fetch o.tickets "
+                    + "join fetch o.user where o.user.id = :userId ", Order.class)
+                    .setParameter("userId", user.getId())
+                    .getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get orders of user " + user.toString(), e);
+        }
```
Comment (dariiabutorina) [inline src/main/java/com/cinema/dao/impl/OrderDaoImpl.java]: Should we catch an exception during the read operations?

File: src/main/java/com/cinema/model/Order.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+@Entity
+@Table(name = "orders")
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToMany
+    private List<Ticket> tickets;
+    private LocalDateTime orderDate;
+    @ManyToOne
```
Comment (dariiabutorina) [inline src/main/java/com/cinema/model/Order.java:22]: Screenshot 2020-10-12 at 13 13 49

File: src/main/java/com/cinema/service/impl/OrderServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    @Inject
+    private OrderDao orderDao;
+
+    @Override
+    public Order completeOrder(List<Ticket> tickets, User user) {
+        Order order = new Order();
+        order.setTickets(tickets);
+        order.setUser(user);
+        order.setOrderDate(LocalDateTime.now());
+        return orderDao.add(order);
```
Comment (dariiabutorina) [inline src/main/java/com/cinema/service/impl/OrderServiceImpl.java]: What should happen to the shopping cart's tickets after completing an order?

**Review by dariiabutorina — CHANGES_REQUESTED** [review]

File: src/main/java/com/cinema/dao/impl/OrderDaoImpl.java (hunk 1 above)
Comment (TimoxaV) [reply src/main/java/com/cinema/dao/impl/OrderDaoImpl.java]: it's not necessary. agree

File: src/main/java/com/cinema/model/Order.java (hunk 2 above)
Comment (TimoxaV) [reply src/main/java/com/cinema/model/Order.java:22]: FetchType.Eager can be used with ManyToOne connection

File: src/main/java/com/cinema/service/impl/OrderServiceImpl.java (hunk 3 above)
Comment (TimoxaV) [reply src/main/java/com/cinema/service/impl/OrderServiceImpl.java]: fixed

File: src/main/java/com/cinema/model/Order.java (hunk 2 above)
Comment (dariiabutorina) [reply src/main/java/com/cinema/model/Order.java:22]: Agree, sorry for confusing

File: src/main/java/com/cinema/dao/ShoppingCartDao.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.cinema.dao;
+
+import com.cinema.model.ShoppingCart;
+import com.cinema.model.User;
+
+public interface ShoppingCartDao {
```
Comment (KseniiaMakarova) [inline src/main/java/com/cinema/dao/ShoppingCartDao.java:6]: let's make one more commit here and see if these classes disappear

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/cinema/dao/ShoppingCartDao.java (hunk 4 above)
Comment (TimoxaV) [reply src/main/java/com/cinema/dao/ShoppingCartDao.java:6]: done

**Pull request closed without merging**